src/lib/dnssd/TxtFields.cpp

```diff
@@ -268,9 +263,12 @@ void FillNodeDataFromTxt(const ByteSpan & key, const ByteSpan & value, CommonRes
     case TxtFieldKey::kSessionActiveThreshold:
         nodeData.mrpRetryActiveThreshold = Internal::GetRetryActiveThreshold(value);
         break;
-    case TxtFieldKey::kTcpSupported:
-        nodeData.supportsTcp = Internal::MakeBoolFromAsciiDecimal(value);
+    case TxtFieldKey::kTcpSupported: {
+        uint8_t support = Internal::MakeU8FromAsciiDecimal(value);
+        nodeData.supportsTcpClient = (support >> 1 & 0x01) == 0x01;
+        nodeData.supportsTcpServer = (support >> 2 & 0x01) == 0x01;
```

Comment:
Please don't write code that makes the reader figure out the order of operations.  This should be doing things more like `(support & 0x01) != 0` or something.

Reply:
+1. Recommend defining an enum with TCP_CLIENT = 1, TCP_SERVER = 2. Then check `(support & (1 << TCP_CLIENT) != 0` for client support and `(support & (1 << TCP_SERVER) != 0` for server support.